src/features/subscription/NotificationsConnectionModal.native.test.tsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react'
+
+import { navigate } from '__mocks__/@react-navigation/native'
+import { NotificationsConnectionModal } from 'features/subscription/NotificationsConnectionModal'
+import { reactQueryProviderHOC } from 'tests/reactQueryProviderHOC'
+import { fireEvent, render, screen, waitFor } from 'tests/utils'
+
+jest.mock('features/profile/pages/NotificationSettings/usePushPermission', () => ({
+  usePushPermission: jest.fn(() => ({
+    pushPermission: 'granted',
+  })),
+}))
+
+const mockDismissModal = jest.fn()
+
+describe('<NotificationsConnectionModal />', () => {
+  it('should render correctly', () => {
+    renderModal(true)
+
+    expect(screen).toMatchSnapshot()
+  })
+
+  it('should dismiss modal on press rightIconButton', () => {
+    renderModal(true)
+
+    const dismissModalButton = screen.getByTestId('rightIcon')
+
+    fireEvent.press(dismissModalButton)
+
+    expect(mockDismissModal).toHaveBeenCalledTimes(1)
+  })
+
+  it('should dismiss modal on press "Créer un compte"', () => {
+    renderModal(true)
+
+    const authButton = screen.getByText('Créer un compte')
+
+    fireEvent.press(authButton)
+
+    expect(mockDismissModal).toHaveBeenCalledTimes(1)
+  })
+
+  it('should go to connection page on press "Créer un compte"', async () => {
```

Comment:
Dans l'app, on dit plutôt `signup` pour l'inscription, et `signin` ou `login` pour la connexion
```suggestion
  it('should go to signup page on press "Créer un compte"', async () => {
```